### PR TITLE
Scopes Component

### PR DESCRIPTION
**Pull request by jasonLaster**
It shows a list of closures / scopes of different types
(functions, blocks, objects).

It also shows a list of scope bindings (variables and arguments).

It's pretty simple right now, but i'm sure will become more complicated
as we go :)

[preview](http://g.recordit.co/dxWoEMUpD6.gif)

### CODE REVIEW

File: public/js/components/Scopes.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  const functionClass = scope.getIn(["function", "class"]);
+
+  const label = displayName || functionClass;
+  return dom.li({ className: "scope-item", key: scope.get("actor") },
+    dom.div({ className: "scope-label" }, (isFirst ? "Local " : "") + label),
+    renderScopeBindings(scope.get("bindings"))
+  );
+}
+
+/* TODO: render binding with variables */
```
Comment (jasonLaster) [inline public/js/components/Scopes.js]: oops, meant render block

File: public/js/components/Scopes.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+function getScopeVariables(scope) {
+  return scope.getIn(["bindings", "variables"]);
+}
+
+function renderArgumentProperty(argumentBinding) {
+  const argumentName = argumentBinding.keySeq().first();
+  const actor = argumentBinding.first().getIn(["value", "actor"]);
+  return dom.li({className: "scope-property-list-item", key: actor},
```
Comment (jlongster) [inline public/js/components/Scopes.js]: Nit: we've been discouraging this system in the devtools codebase. We prefer:

```
dom.li(
  { className: "scope", ... },
  ...
)
```

Generally it's easier to add more properties and looks better when there are more. Also some editors (like mine) want to align the arguments. Also, space after and before curly braces.

**Comment by jlongster** [comment]
Looks like a great start.

**Comment by jasonLaster** [comment]
merged!

**Pull request closed without merging**